File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/03.Conditional-Statements-Advanced/02-exercise/04.fishingBoat.js
function fishingBoat(input) {

    const springPrice = 3000;
    const summerPrice = 4200;
    const autumnPrice = 4200;
    const winterPrice = 2600;

    const budget = Number(input[0]);
    const season = input[1];
    const fishermen = Number(input[2]);

    let finalPrice = 0;

    switch (season) {
        case "Spring": finalPrice = springPrice; break;
        case "Summer": finalPrice = summerPrice; break;
        case "Autumn": finalPrice = autumnPrice; break;
        case "Winter": finalPrice = winterPrice; break;
    }

    if (fishermen <= 6) {
        finalPrice *= 0.90;
    } else if (fishermen > 6 && fishermen <= 11) {
        finalPrice *= 0.85;
    } else if (fishermen > 11) {
        finalPrice *= 0.75;
    }

    if (fishermen % 2 === 0 && season !== "Autumn") {
        finalPrice *= 0.95;
    }

    const difference = Math.abs(budget - finalPrice).toFixed(2);

    if (budget >= finalPrice) {
        console.log(`Yes! You have ${difference} leva left.`);
    } else {
        console.log(`Not enough money! You need ${difference} leva.`);
    }
}
fishingBoat(["2000", "Winter", "13"]);

File: Python_Basics/Homeworks-And-Labs/03.conditional_statements_advanced/02.exercise/04.fishing_boat.py
springPrice = 3000
summerPrice = 4200
autumnPrice = 4200
winterPrice = 2600

budget = int(input())
season = input()
fishermen = int(input())

finalPrice = 0

if season == 'Spring':
    finalPrice = springPrice
elif season == 'Summer':
    finalPrice = summerPrice
elif season == 'Autumn':
    finalPrice = autumnPrice
elif season == 'Winter':
    finalPrice = winterPrice

if fishermen <= 6:
    finalPrice *= 0.90
elif fishermen > 6 and fishermen <= 11:
    finalPrice *= 0.85
elif fishermen > 11:
    finalPrice *= 0.75

if fishermen % 2 == 0 and season != "Autumn":
    finalPrice *= 0.95

difference = f'{abs(budget - finalPrice):.2f}'

if budget >= finalPrice:
    print(f'Yes! You have {difference} leva left.')
else:
    print(f'Not enough money! You need {difference} leva.')
